client/src/js/reducers/DocumentsReducer.js

```diff
@@ -10,6 +10,15 @@ export default (state = [], action) => {
       return [...state, action.payload];
     case types.DELETE_DOCUMENT:
       return state.filter(doc => action.id !== doc.id);
+    case types.UPDATE_DOCUMENT:
+      console.log(state, 'new update state');
+      console.log(action.updatedDocument, 'new real state');
+      return state.map((doc) => {
```

Comment:
Do you really need to loop through the state to be able to update the particular doc that got updated? What happens if there are several documents, Is it possible to find a way to pass the index of the updated document to the reducer?